### PR TITLE
Update gitignore and fix warnings

### PR DESCRIPTION
**Pull request by wesbland**
## Pull Request Description

Fix the gitignore and some recently introduced warnings.

This PR also updates the name of the `code-cleanup.sh` script to `code-cleanup.bash` to better match the convention in MPICH and MPICH-related projects.

## Expected Impact

none

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

**Comment by wesbland** [comment]
@hzhou I can't assign reviewers on this one, but I think it goes to you. It's nice and small. 😄

**Review by hzhou — APPROVED** [review]

**Comment by wesbland** [comment]
Sorry. Hold on a minute before merging. I may have just found another thing to fix.

**Comment by wesbland** [comment]
Ok. Seems to be an MPICH build issue. Not a yaksa one. Good to go when the tests finish.

**Pull request merged**
